hal/src/sercom/v2/uart/mod.rs

```diff
@@ -0,0 +1,1528 @@
+    // All flags are valid for a Duplex UART
+    const FLAG_MASK: u8 = DUPLEX_FLAG_MASK;
+
+    // All status flags are valid for a Duplex UART
+    const STATUS_MASK: u16 = DUPLEX_STATUS_MASK;
+}
+impl Receive for Duplex {}
+impl Transmit for Duplex {}
+
+/// Marker type representing a UART that can only receive
+pub enum Rx {}
+impl Sealed for Rx {}
+impl Capability for Rx {
+    // Available interrupt flags for a RX half-UART
+    const FLAG_MASK: u8 = RX_FLAG_MASK;
+
+    // Available status flags for a RX half-UART
+    const STATUS_MASK: u16 = RX_STATUS_MASK;
+}
+impl Receive for Rx {}
+impl Simplex for Rx {}
+
+/// Marker type representing a UART that can only transmit
+pub enum Tx {}
+impl Sealed for Tx {}
+impl Capability for Tx {
+    // Available interrupt flags for a TX half-UART
+    const FLAG_MASK: u8 = TX_FLAG_MASK;
+
+    // Available status flags for a TX half-UART
+    const STATUS_MASK: u16 = TX_STATUS_MASK;
+}
+impl Transmit for Tx {}
+impl Simplex for Tx {}
+
+/// Marker type representing the Rx half of a  [`Duplex`] UART
+pub enum RxDuplex {}
+impl Sealed for RxDuplex {}
+impl Capability for RxDuplex {
+    // Available interrupt flags for a RX half-UART
+    const FLAG_MASK: u8 = RX_FLAG_MASK;
+
+    // Available status flags for a RX half-UART
+    const STATUS_MASK: u16 = RX_STATUS_MASK;
+}
+impl Receive for RxDuplex {}
+
+/// Marker type representing a the Tx half of a [`Duplex`] UART
+pub enum TxDuplex {}
+impl Sealed for TxDuplex {}
+impl Capability for TxDuplex {
+    // Available interrupt flags for a TX half-UART
+    const FLAG_MASK: u8 = TX_FLAG_MASK;
+
+    // Available status flags for a TX half-UART
+    const STATUS_MASK: u16 = TX_STATUS_MASK;
+}
+impl Transmit for TxDuplex {}
+
+//=============================================================================
+// Config
+//=============================================================================
+
+/// A configurable, disabled UART peripheral
+///
+/// This `struct` represents a configurable UART peripheral in its disabled
+/// state. It is generic over the set of [`Pads`] and [`CharSize`].
+/// Upon creation, the [`Config`] takes ownership of the
+/// [`Sercom`] and resets it, returning it configured as an UART peripheral
+/// with a default configuration:
+///
+/// * [`EightBit`]
+/// * No parity
+/// * One stop bit
+/// * LSB-first
+///
+/// [`Config`] uses a builder-pattern API to configure the peripheral,
+/// culminating in a call to [`enable`], which consumes the [`Config`] and
+/// returns enabled [`Uart`]. The [`enable`] method is
+/// restricted to [`ValidConfig`]s.
+///
+/// [`enable`]: Config::enable
+pub struct Config<P, C = EightBit>
+where
+    P: ValidPads,
+    C: CharSize,
+{
+    registers: Registers<P::Sercom>,
+    pads: P,
+    chsize: PhantomData<C>,
+    freq: Hertz,
+}
+
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+type Clock = pac::PM;
+
+#[cfg(feature = "min-samd51g")]
+type Clock = pac::MCLK;
```

Comment:
I forgot about this problem. Simple enough solution though. Hopefully we can unify this with `clock::v2`.